src/shims/unix/foreign_items.rs

```diff
@@ -553,6 +555,103 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
                 let result = this.nanosleep(req, rem)?;
                 this.write_scalar(Scalar::from_i32(result), dest)?;
             }
+            "sched_getaffinity" => {
+                // Currently this function does not exist on all Unixes, e.g. on macOS.
+                if !matches!(&*this.tcx.sess.target.os, "linux" | "freebsd" | "android") {
+                    throw_unsup_format!(
+                        "`sched_getaffinity` is not supported on {}",
+                        this.tcx.sess.target.os
+                    );
+                }
+
+                let [pid, cpusetsize, mask] =
+                    this.check_shim(abi, Abi::C { unwind: false }, link_name, args)?;
+                let pid = this.read_scalar(pid)?.to_u32()?;
+                let cpusetsize = this.read_target_usize(cpusetsize)?;
+                let mask = this.read_pointer(mask)?;
+
+                let thread_id = match pid {
+                    0 => this.active_thread(),
+                    _ => ThreadId::from(pid),
```

Comment:
I think this is the wrong ID... `ThreadId` corresponds to the pthtread ID, not the kernel PID/TID. Miri currently does not have a notion of PID/TID for threads (see https://github.com/rust-lang/miri/issues/3730).

Would it suffice to only support pid 0 for now?

Reply:
On MacOS, `man getpid` documents

>      getpid() returns the process ID of the calling process.  The ID is
>      guaranteed to be unique and is useful for constructing temporary file
>      names.

So I don't think always returning 0 would be a valid implementation.

Reply:
(not that I have any particular advice on what else to do)

Reply:
We're not returning a PID here, we are taking a PID as argument, and 0 is specified to mean "current thread". So I think we can declare that we don't support any other value.

Reply:
Ah! Yes.

Reply:
> Would it suffice to only support pid 0 for now? The case of setting/getting another thread's mask anyway seems to be untested.

Yes, that covers all actual rust examples I've seen so far, and it's untested exactly because `getttid` is not implemented currently.